openquake/hazardlib/source/simple_fault.py

```diff
@@ -143,13 +149,34 @@ def iter_ruptures(self):
                 for first_col in xrange(num_rup_along_length):
                     mesh = whole_fault_mesh[first_row: first_row + rup_rows,
                                             first_col: first_col + rup_cols]
-                    hypocenter = mesh.get_middle_point()
-                    surface = SimpleFaultSurface(mesh)
-                    yield ParametricProbabilisticRupture(
-                        mag, self.rake, self.tectonic_region_type, hypocenter,
-                        surface, type(self),
-                        occurrence_rate, self.temporal_occurrence_model
-                    )
+
+
+                    if self.hypo_list.size == 1:
+
+                        hypocenter = mesh.get_middle_point()
+                        occurrence_rate_hypo = occurrence_rate 
+                        surface = SimpleFaultSurface(mesh)
+
+                        yield ParametricProbabilisticRupture(
+                            mag, self.rake, self.tectonic_region_type, hypocenter,
+                            surface, type(self),
+                            occurrence_rate_hypo, self.temporal_occurrence_model
+                        )                  
+                    else:
+                        for index in range(len(self.hypo_list)):
+
+                            surface = SimpleFaultSurface(mesh)
+                            hypocenter = surface.get_hypo_location(mesh, self.rupture_mesh_spacing, \
+                                                               (self.hypo_list[index][0], self.hypo_list[index][1]))
```

Comment:
"Can we just pass self.hypo_list[index]?
Or even better loop directly on the hypo_list?"

What do you mean loop directly on the hypo_list?